### PR TITLE
UW-523 Support Jinja2 template loading

### PR DESCRIPTION
**Pull request by maddenp-noaa**
**Synopsis**

Provide support for informing Jinja2 about paths to search for templates to load via `import` et al. template expressions. By default, the directory containing the primary template will be searched for loadable templates (when `-i` / `--input-file` is specified). A `--search-path` CLI option / `searchpath` API argument can be used to suppress this default and provide a set of colon-separated paths to search.

**Type**

- [x] Documentation
- [x] Enhancement (adds a new functionality)

**Impact**

- [x] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

File: src/uwtools/config/jinja2.py [hunk 1]
```diff
@@ -30,19 +22,44 @@ class J2Template:
...
-            else self._load_file(template_source)
-        )
         self._template_source = template_source
+        self._j2env = Environment(
+            loader=FileSystemLoader(
+                searchpath=searchpath.split(":")
+                if searchpath
+                else self._template_source.parent
+                if isinstance(self._template_source, Path)
+                else []
```
Comment (maddenp-noaa) [inline src/uwtools/config/jinja2.py:45]: Note the precedence for the search path:
1. Use the explicitly specified search path(s) if provided;
2. Use the directory containing the input template if possible;
3. Use no search path, as none can be defined.

File: src/uwtools/config/jinja2.py [hunk 2]
```diff
@@ -72,37 +89,9 @@ def undeclared_variables(self) -> Set[str]:
...
-    def _load_string(self, template: str) -> Template:
-        """
-        Load the Jinja2 template from the string provided.
-
-        :param template: An in-memory Jinja2 template.
-        :return: The Jinja2 template object.
-        """
-        self._j2env = Environment(loader=BaseLoader())
-        _register_filters(self._j2env)
-        return self._j2env.from_string(template)
```
Comment (maddenp-noaa) [inline src/uwtools/config/jinja2.py:104]: These methods are no longer needed.

**Review by WeirAE — COMMENTED** [review]
Pending confirmation from the user, this was an excellent solution.

**Review by christinaholtNOAA — APPROVED** [review]
Looks good to me! Nice turn around speed on the user report/request.

**Pull request merged**